docs/topics/forms/modelforms.txt

```diff
@@ -1311,6 +1311,51 @@ Then when you create your inline formset, pass in the optional argument
     >>> author = Author.objects.get(name="Mike Royko")
     >>> formset = BookFormSet(instance=author)
 
+.. _inline-formset-interaction-with-prefetching:
+
+Interaction with prefetching
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 5.1
+
+    The ``get_instance_by_queryset_index`` and ``prepare_queryset`` hooks were added.
+
+You'll need to create a subclass to take advantage of
```

Comment:
Reading this section, it's not clear for me what we want to achieve. We should start from describing what we're trying to do here.